### ISSUE TITLE
It is recommended to update the dependent version

### ISSUE THREAD

**Issue by CQBoyBrand** [SimulatedGREG/electron-vue#855]
It is recommended to update the dependent version,for example:electron

**Comment by linuxjuggler** [comment]
You should do that as each version will fix something, but you will need to do that carefully.

My advice is to upgrade each package alone then test your code if everything went without any problem move to the next package, otherwise, keep the same version and move to the next package.

right now you will get `"electron": "^2.0.4",` out of the box, but the latest version which has a lot of fixes and improvements is `"electron": "^5.0.1",`.

**Comment by BorysTyminski** [comment]
@linuxjuggler what is most convenient way of updating electron and node version which are shipped with `vue-electron`?

**Comment by linuxjuggler** [comment]
@BorysTyminski node is not shipped with `electron-vue`, so it's up to you how you upgrade node, for me I use [nvm](https://github.com/nvm-sh/nvm) which works on both Linux and Mac.

For the modules, you can use `npm outdated` command which will list all the outdated modules that you need to install, so using `npm install electron@latest` for example, will upgrade electron to the latest version, but I advise you to test your code with each library update you perform.

**Comment by BorysTyminski** [comment]
@linuxjuggler that's weird about node because when I run `node -v` in terminal i get:

![image](https://user-images.githubusercontent.com/40339232/58465467-11e6c980-8138-11e9-906a-11df76d4168e.png)

but when I open dev tools within electron window and call `process.versions` in console:

![image](https://user-images.githubusercontent.com/40339232/58465525-2b881100-8138-11e9-9560-b6892fa6227c.png)

and because of that reason I'm not able to use worker within my electron project.

**Comment by linuxjuggler** [comment]
you should upgrade `electron` itself

```shell
$ npm i -D electron@latest
# Electron   5.0.2
# Node       12.0.0
# Chromium   73.0.3683.121
```

**Comment by BorysTyminski** [comment]
I run this command `npm install electron@latest` and now basically my app is launching but not displaying anything and there are two errors in electron window devtools:

> Uncaught ReferenceError: require is not defined at (index):1
> Uncaught ReferenceError: module is not defined at renderer.js:1

but `const Worker = require('worker_threads');` is finally not telling me that it can not find this module.

Also now `process.versions` in electron window devtools throw error:

> Uncaught ReferenceError: process is not defined

**Comment by linuxjuggler** [comment]
try to use node 11 instead of node 12, I remember facing small issues when I updated to node 12 and that's why am stick with node 11 on my system 
```
➜ node --version
v11.15.0
```

**Comment by linuxjuggler** [comment]
Just updated to version 5.0.2 and it is working

![Screen Shot 2019-05-28 at 12 51 34 PM](https://user-images.githubusercontent.com/27624/58469034-634a8500-8147-11e9-8fb8-b975120fa1e0.png)

**Comment by BorysTyminski** [comment]
Okey so as I said I created fresh project with electron-vue:

> vue init simulatedgreg/electron-vue electron-vue-issue

then I updated electron with

> npm i -D electron@latest

then I double checked my node version:

> node -v
> v11.15.0

then I run project with

> npm run dev

Result is that I have same errors in devtools of electron window as in my original project:

> Uncaught ReferenceError: require is not defined at (index):1
> Uncaught ReferenceError: module is not defined at renderer.js:1

and also this one

> Uncaught ReferenceError: process is not defined at (index):1

which can be solved by removing few lines from `index.ejs` as seen in #871.

I've created [demo repository on github](https://github.com/BorysTyminski/electron-vue-issue). I would very appreciate and would be very thankful because it will help me a lot if you manage to spot the difference between your successfully working project and this issue demo project.

**Comment by linuxjuggler** [comment]
I tried your code, and it also failed, but I noticed something maybe the cause of the issue, and to be honest I don't have the full time now to test it, perhaps in the evening, I'll try to do more digging.

For some reason, in your new code the `renderer process` starts before the `main process` meanwhile in my code the main will compile before the renderer, not sure if this has any effect on the project or not.

![Screen Shot 2019-05-28 at 5 22 49 PM](https://user-images.githubusercontent.com/27624/58485885-8a1bb200-816d-11e9-8f39-96182250082c.png)

**Comment by BorysTyminski** [comment]
This is so weird. It's hard to say that it's my code it's just this boilerplate with updated electron to latest version nothing more. But I found out that version 4.3.2 of electron is working just fine and have NodeJS version which allow me to create node-worker. 

This electron-vue is starting making me a headache, it suppose to save time and I literally wasted so many hours today trying to do sipmple things...

**Comment by linuxjuggler** [comment]
@BorysTyminski I found the problem which cause the issue:

in your `src/main/index.js` file you will need to add the following to the `BrowserWindow` function:

```
    webPreferences: {
      nodeIntegration: true,
    }, 
```

> nodeIntegration Boolean (optional) - Whether node integration is enabled. Default is false.

So the new code to create a new `BrowserWindow` is:

```
  mainWindow = new BrowserWindow({
    height: 563,
    useContentSize: true,
    width: 1000,
    webPreferences: {
      nodeIntegration: true,
    },    
  })
```

Anyway, I advice you read more about the option `webPreferences` at the [documentation](https://electronjs.org/docs/api/browser-window)

**Comment by heronggen** [comment]
I have updated electron to latest version (v5.0.4) :

then I updated electron 

`npm i -D electron@latest`

in your src/main/index.js file you will need to add the following to the BrowserWindow function:

    webPreferences: {
      nodeIntegration: true,
    }, 

then changed electron-builder version in your package.json
 ` "electron-builder": "^20.41.0",`

So the new  you run `yarn install` again

![TIM截图20190619175737](https://user-images.githubusercontent.com/32705514/59756275-f9437c80-92bb-11e9-942b-3ec36d5b8325.png)

**Comment by yyman001** [comment]
for the time at 2019-07-30, electron last verison 6.0, support upgrading to this version?

**Comment by linuxjuggler** [comment]
@yyman001 yes it will work like a charm, you need to add `app.setAppLogsPath();` before using `app.getPath('logs')`

other than that not much changed. (at least in my case)

**Comment by yyman001** [comment]
@linuxjuggler Does dependency cause problems ? 
The default installation is 2.x version, I don't know if this version is too old, I don't know what will happen if I upgrade to the new version?
    thanks you!

**Comment by linuxjuggler** [comment]
@yymna001 upgrading from 2 to 6 is tricky but if you followed the instructions here you will get it done in no time (am talking about electron in specific here). For sure the other dependency needs small amount of work to get upgraded so I advise you to try it in an empty/new project.

**Comment by MisterJD** [comment]
After an upgrade from electron version 5.0.8 to 6.0.2 no window is shown up anymore in dev mode (build is working like a charm). Is someone else facing this problem? #871 already applied.

**Comment by linuxjuggler** [comment]
@MisterJD can you provide more information about the error messages you are getting?

**Comment by MisterJD** [comment]
@linuxjuggler unfortunately, there a none. everything seems fine. electron processes running in the background, but no window is shown up (even no developer console). then I have to cancel the batch process manuelly per console. Same behaviour occur if I try the new boilerplate from #888. Already upgraded node.js to the latest version for testing, nothing changed. If I switch back to electron 5.0.9, dev command is working again. I am really missing a starting point here for digging in. I keep searching.

**Comment by linuxjuggler** [comment]
try to upgrade to version 6.0.0 first, and be aware of this small change https://github.com/SimulatedGREG/electron-vue/issues/855#issuecomment-517864370 which may be the problem

**Comment by MisterJD** [comment]
mh, strange. https://github.com/SimulatedGREG/electron-vue/issues/855#issuecomment-517864370 already applied. okay, then it seems this is specfic to my dev environment. thanks for your help & effort @linuxjuggler . I have to search the error on my end.

**Comment by linuxjuggler** [comment]
@MisterJD if you are on a mac you can check the `Console.app` and check if there are any logs there related to your app.

**Comment by f2rge** [comment]
@MisterJD (electron version 5.0.6 to 6.0.7) I have the same problem. It cannot be resolved.

**Comment by MisterJD** [comment]
@f2rge Under Windows I have to delete the `DevTools Extensions` File in the folder `C:\Users\%user%\AppData\Roaming\Electron` everytime on startup. Don't ask me why, it seems it have something to do with the installation of the Vue extension.

**Comment by InsaneCake** [comment]
Ran into same issue as @MisterJD after updating electron from 2.0.x to 7.1.5 (latest at this moment) in clear project.
[This solution](https://github.com/SimulatedGREG/electron-vue/issues/855#issuecomment-529209390) combined with [this](https://github.com/SimulatedGREG/electron-vue/issues/855#issuecomment-503002257) works, so i commented installing `vue-devtools` in `main/index.dev.js` and i dont have to delete file every time (by cost of vue devtools, ofcourse)

![image](https://user-images.githubusercontent.com/14079980/70998638-d4a05400-20e8-11ea-89b2-6b2fac8c427d.png)

but it works only in dev

this code in `main/index.js`:
```
console.log(app.isReady())
app.on('ready', () => {
  console.log(app.isReady())
  createWindow()
})
```
in dev run logs `false`, then `true`
but, if i run built (by build:dir) project from console, it only logs `false`
Seems like `ready` does not firing

Any thoughts?

**Comment by InsaneCake** [comment]
I found solution, that works for me and made a sample [repo](https://github.com/InsaneCake/electron-vue-template) with updated electron (7.1.7)
In [this commit](https://github.com/InsaneCake/electron-vue-template/commit/f983a6911fa3f19aa64f76cc425d95cf21bb03ed) all changes, that have been made compared to the new project

**Comment by sabifa** [comment]
> I found solution, that works for me and made a sample [repo](https://github.com/InsaneCake/electron-vue-template) with updated electron (7.1.7)
> In [this commit](https://github.com/InsaneCake/electron-vue-template/commit/f983a6911fa3f19aa64f76cc425d95cf21bb03ed) all changes, that have been made compared to the new project

After I tried to update my project to electron 7 which didn't work, I tried your sample repo. Unfortunately the electron app does not start at all, main process has no errors. Do I have to run a specific node version? 
I'm currently on v11.15.0

**Comment by InsaneCake** [comment]
> I managed to resolve this problem by deleting the cache folder of electron located at C:\Users\AppData\Roaming\Electron it starts flawlessly running Electron 7.1.7 now!

Yes, you should keep an aye on this folder, especially on file "DevTools Extensions" in it.
I mention it in readme (not in clearest way):
>if you need vue devtools extension, then uncomment strings 14 - 21 in src/main/index.dev.js, but then you have to delete "DevTools Extensions" file in C:\Users\%USER%\AppData\Roaming\Electron before every dev launch.

If you run another project with devtools installation enabled, this file will be created again and broke dev startup of updated project.
I will update repo readme later.

**Comment by hetz** [comment]
I get same problem, electron today release  8.0 https://www.electronjs.org/releases/stable .

**Comment by Brouilles** [comment]
Same @hetz

**Comment by underaft** [comment]
I'm having a lot of problems trying to upgrade stuff... Like electron 8 and some VUE packages that are also outdated... I think this repo needs a revision 😢
